### PR TITLE
Auto-update tracking graypaper releases

### PR DESCRIPTION
**Pull request by skoszuta**

### CODE REVIEW

**Review by tomusdrw — APPROVED** [review]

File: .github/workflows/auto-update.yml [hunk 1]
```diff
@@ -50,21 +50,24 @@ jobs:
...
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          tag_name: ${{ steps.graypaper_update.outputs.VERSION }}
-          release_name: Release ${{ steps.graypaper_update.outputs.VERSION }}
-          body: Bumped Graypaper version to ${{steps.graypaper_update.outputs.VERSION}}
-          draft: false
-          prerelease: false
+        run: |
+          gh release create ${{ steps.graypaper_update.outputs.VERSION }} \
```
Comment (tomusdrw) [inline .github/workflows/auto-update.yml:70]: Shouldn't we wait with the release till the merge is done? What code will be referenced in the release? Is it going to be the PR content or what was merged to the main branch?

File: .github/workflows/auto-update.yml [hunk 2]
```diff
@@ -50,21 +50,24 @@ jobs:
...
+          git config --global user.email $BOT_EMAIL
+      - name: Create a PR and merge changes to the main branch
+        run: |
+          git checkout -b feature/${{steps.graypaper_update.outputs.VERSION}}
+          git add .
+          git commit -m "Bumped Graypaper to ${{steps.graypaper_update.outputs.VERSION}}"
+          git push --set-upstream origin feature/${{steps.graypaper_update.outputs.VERSION}}
+          gh pr create -B main -H feature/$VERSION \
+            --title "Bump Graypaper to ${{steps.graypaper_update.outputs.VERSION}}" --body ""
+          gh pr merge --merge --delete-branch
```
Comment (tomusdrw) [inline .github/workflows/auto-update.yml:67]: I guess this will create a merge commit. Can we rebase instead?

**Review by tomusdrw — APPROVED** [review]

File: .github/workflows/auto-update.yml (hunk 1 above)
Comment (skoszuta) [reply .github/workflows/auto-update.yml:71]: afaik `gh pr merge` is synchronous therefore the release should only happen when merge was successful.

As for what code will be referenced this is a quote from `gh release` docs:

> If a matching git tag does not yet exist, one will automatically get created from the latest state of the default branch

File: .github/workflows/auto-update.yml (hunk 2 above)
Comment (skoszuta) [reply .github/workflows/auto-update.yml:68]: yes we can, of course

**Pull request merged**